src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -296,32 +296,31 @@ static void restore_eliminated_locks(JavaThread* thread, GrowableArray<compiledV
                                                      exec_mode, realloc_failures);
       deoptimized_objects = deoptimized_objects || relocked;
 #ifndef PRODUCT
-      if (PrintDeoptimizationDetails) {
-        ResourceMark rm;
-        stringStream st;
+      LogMessage(deoptimization) lm;
```

Comment:
Drive-by: Does this not already incur costs? Does LogMessage not contain an internal buffer it allocates?